### PR TITLE
chore: Updated API reference comments to tsdoc.org standards

### PR DESCRIPTION
**Pull request by hiranya911**
See https://tsdoc.org/ for details on syntax.

* Changed all `@return` annotations to `@returns`.
* Changed all markdown and other links to `{@link}`.
* Changed relative URLs to absolute URLs.
* Removed many instances of old jsdoc style parameter and return type declarations.

### CODE REVIEW

File: src/app/core.ts [hunk 1]
```diff
@@ -20,29 +20,30 @@ import { Agent } from 'http';
...
-   * See [Initialize the SDK](/docs/admin/setup#initialize_the_sdk) for detailed
-   * documentation and code samples.
+   * See {@link https://firebase.google.com/docs/admin/setup#initialize_the_sdk | Initialize the SDK}
+   * for detailed documentation and code samples.
    */
   credential?: Credential;
 
   /**
-   * The object to use as the [`auth`](/docs/reference/security/database/#auth)
+   * The object to use as the {@link https://firebase.google.com/docs/reference/security/database/#auth | auth}
```
Comment (egilmorez) [inline src/app/core.ts:37]: Do you know if it's possible to keep the code styling?  

Not required, but nice to have.

File: src/messaging/messaging.ts [hunk 2]
```diff
@@ -134,9 +134,9 @@ function mapRawResponseToDevicesResponse(response: object): MessagingDevicesResp
 /**
  * Maps a raw FCM server response to a MessagingDeviceGroupResponse object.
  *
- * @param {object} response The raw FCM server response to map.
```
Comment (egilmorez) [inline src/messaging/messaging.ts:137]: Are these values in the curly brackets types?  Not sure if they are important enough to try and transmit to the new format.

**Review by egilmorez — APPROVED** [review]
Thanks!

File: src/messaging/messaging.ts (hunk 2 above)
Comment (hiranya911) [reply src/messaging/messaging.ts:137]: This is the old jsdoc way of documenting types. That's not necessary with TypeScript and Typedoc/tsdoc.

File: src/app/core.ts (hunk 1 above)
Comment (hiranya911) [reply src/app/core.ts:37]: Haven't been able to get this to work. I've asked tsdoc.org community to see if I've missed something.

**Review by lahirumaramba — APPROVED** [review]

**Pull request merged**